BinaryTree.java

```diff
@@ -0,0 +1,165 @@
+package com.katermar;
+
+/**
+ * Created by katermar on 5/23/2017.
+ */
+public class BinaryTree {
+    TreeNode root;
+
+    /*Inner class for Binary Tree node
+    * @field value stores integer value
+    * @field left, right store left and right subtrees*/
+    public static class TreeNode {
+        private int value;
+        private TreeNode left, right;
+
+        TreeNode(int item) {
+            value = item;
+            left = right = null;
+        }
+
+        public int getValue() {
+            return value;
+        }
+
+        public TreeNode getLeft() {
+            return left;
+        }
+
+        @Override
+        public String toString() {
+            return "TreeNode{" +

```

Comment:
Alternatively you could use String.format, which might be a bit less efficient, but results in a nicer code.